contracts/KeyManager.sol

```diff
@@ -51,6 +51,11 @@ contract KeyManager {
       "Key is revoked"
     );
 
+    require(
+      key != addressToKey(address(this)) && purpose != MANAGEMENT,
+      "Own address can not be a management key"
+    );
+
     if (!keyHasPurpose(key, purpose)) {
       _keys[key].keyType = keyType;
```

Comment:
Currently, it is possible to add a key with one purpose and add another purpose later on.
```suggestion
       if (_keys[key].purposes.length == 0) {
      _keys[key].keyType = keyType;
      }
```

But I think it should be not possible to change the `keyType` after the key has been already added.